src/Datadog.Trace/Vendors/Newtonsoft.Json/Serialization/JsonPropertyCollection.cs

```diff
@@ -144,6 +144,7 @@ public JsonProperty GetClosestMatchProperty(string propertyName)
             return property;
         }
 
+#if !NETCOREAPP
```

Comment:
How come this change was needed?

Reply:
Looks like TryGetValue was added to the KeyedCollection type at some point: https://source.dot.net/#System.ObjectModel/System/Collections/ObjectModel/KeyedCollection.cs,95

Reply:
Maybe I should mark the method as `new` instead, to avoid any behavior change

Reply:
Any more follow up thought on this? I like not introducing a potential behavior change.

Reply:
Looks like KeyedCollection can use either a list or a dictionary for internal storage: https://source.dot.net/#System.ObjectModel/System/Collections/ObjectModel/KeyedCollection.cs,95
I think it's best to rely on the base implementation which is aware of that